api/catalog/api/utils/validate_images.py

```diff
@@ -28,6 +30,28 @@ def _get_expiry(status, default):
     return config(f"LINK_VALIDATION_CACHE_EXPIRY__{status}", default=default, cast=int)
 
 
+async def _head(url: str, session: aiohttp.ClientSession):
+    try:
+        async with session.head(url, timeout=2, allow_redirects=False) as response:
+            return url, response.status
+    except aiohttp.ClientError as exception:
+        _validation_failure(exception)
+        return url, -1
+
+
+# https://stackoverflow.com/q/55259755
+@async_to_sync
+async def _make_head_requests(urls: list[str]):
```

Comment:
It is a coroutine, but does the return type need to be explicitly set as a coroutine? This makes it seem like you do not need to wrap the return type annotation in `Awaitable` as it's implied by it being on an async function: https://docs.python.org/3/library/typing.html#callable

I think it can just be as you suggested, the type of the result of the coroutine.

Reply:
Done in https://github.com/WordPress/openverse-api/pull/1027/commits/582d563ada7def1b8ded7b0c80d3b994fe22deda